src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2849,6 +2874,58 @@ FlowAnalysisAnnotations removeInapplicableNotNullWhenSense(ParameterSymbol param
             }
         }
 
+        private static TypeWithAnnotations ApplyLValueAnnotations(TypeWithAnnotations declaredType, FlowAnalysisAnnotations flowAnalysisAnnotations)
+        {
+            if ((flowAnalysisAnnotations & FlowAnalysisAnnotations.DisallowNull) != 0)
+            {
+                var typeSymbol = declaredType.Type;
+                if (declaredType.NullableAnnotation.IsNotAnnotated() || (typeSymbol.IsValueType && !typeSymbol.IsNullableType()))
+                {
+                    return declaredType;
+                }
+
+                return TypeWithAnnotations.Create(typeSymbol, NullableAnnotation.NotAnnotated, declaredType.CustomModifiers);
+            }
+            else if ((flowAnalysisAnnotations & FlowAnalysisAnnotations.AllowNull) != 0)
+            {
+                var typeSymbol = declaredType.Type;
+                if (declaredType.NullableAnnotation.IsAnnotated() || (typeSymbol.IsValueType && typeSymbol.IsNullableType()))
+                {
+                    return declaredType;
+                }
+
+                return TypeWithAnnotations.Create(typeSymbol, NullableAnnotation.Annotated, declaredType.CustomModifiers);
+            }
+
+            return declaredType;
+        }
+
+        private static TypeWithAnnotations ApplyRValueAnnotations(TypeWithAnnotations declaredType, FlowAnalysisAnnotations flowAnalysisAnnotations)
+        {
+            if ((flowAnalysisAnnotations & FlowAnalysisAnnotations.NotNull) == FlowAnalysisAnnotations.NotNull)
```

Comment:
>== FlowAnalysisAnnotations.NotNull [](start = 76, length = 34)

The comparison style should be consistent between this method and the one above.

Reply:
I'll change, but the reason for this comparison style is that `NotNull` is two bits, whereas `DisallowNull` is only one bit (so simpler comparison works ok there)

---
In reply to: [289486556](https://github.com/dotnet/roslyn/pull/35955#discussion_r289486556) [](ancestors = 289486556)